### PR TITLE
fix: silence compiler warning in tbb header about old-style casts.

### PR DESCRIPTION
**Pull request by goetzgaycken**
Silence warnings emitted by GCC 13.1 about old-style c-casts in some headers of tbb 2021.13.

--- END COMMIT MESSAGE ---

Any further description goes here, @-mentions are ok here!

### CODE REVIEW

File: Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp [hunk 1]
```diff
@@ -36,8 +36,10 @@
 #include <string>
 #include <variant>
 #include <vector>
-
+#pragma GCC diagnostic push
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp:39]:
```suggestion

#pragma GCC diagnostic push
```

**Comment by paulgessinger** [comment]
@goetzgaycken Can you copy the warning here for reference?

**Comment by goetzgaycken** [comment]
> @goetzgaycken Can you copy the warning here for reference?

```
[ 96%] Building CXX object Examples/Algorithms/TrackFinding/CMakeFiles/ActsExamplesTrackFinding.dir/src/TrackFindingAlgorithmFunction.cpp.o
In file included from /cvmfs/projects.cern.ch/intelsw/oneAPI/linux/x86_64/2025/tbb/2022.0/env/../include/tbb/../oneapi/tbb/combinable.h:22,
                 from /cvmfs/projects.cern.ch/intelsw/oneAPI/linux/x86_64/2025/tbb/2022.0/env/../include/tbb/combinable.h:17,
                 from /data/goetz/src/ACTS/source/acts-standalone/Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp:40,
                 from /data/goetz/src/ACTS/source/acts-standalone/Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp:9:
/cvmfs/projects.cern.ch/intelsw/oneAPI/linux/x86_64/2025/tbb/2022.0/env/../include/tbb/../oneapi/tbb/enumerable_thread_specific.h: In member function 'tbb::detail::d1::enumerable_thread_specific<T, Allocator, ETS_key_type>::value_type& tbb::detail::d1::enumerable_thread_specific<T, Allocator, ETS_key_type>::local(bool&)':
/cvmfs/projects.cern.ch/intelsw/oneAPI/linux/x86_64/2025/tbb/2022.0/env/../include/tbb/../oneapi/tbb/enumerable_thread_specific.h:894:21: warning: use of old-style cast to 'T*'  -Wold-style-cast]
  894 |         return *(T*)ptr;
      |                     ^~~

```

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [6b31eb2a887fcf01a7b04681e2f1bec185c3dd87]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920723)||
|:green_circle: | [run_art_test: test_run4_acts_vertex_PU200](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920722)||
|:green_circle: | [run_art_test: test_run4_ttbar_PU200](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920721)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920720)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920719)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920718)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920717)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920716)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920715)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920714)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920713)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflowHgtd](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920712)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920711)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920710)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920709)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920708)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920707)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920706)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920705)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920704)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920703)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920702)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920701)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920700)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920699)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920698)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920697)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920696)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920695)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920694)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920693)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920692)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920691)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920690)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920689)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/45920688)||

**Pull request merged**